### PR TITLE
BUGFIX: Add dimensions to Workspace module

### PR DESCRIPTION
**Pull request by pKallert**
resolves: #3470 

This adds Dimension handling to the current workspaces module. 

Currently the Module only displays changes in different languages as one change. This is because the node path is the same in different languages after they are copied.

**Checklist**

- [x] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [x] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [x] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [x] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.Neos/Classes/Controller/Module/Management/WorkspacesController.php
Comment (markusguenther) [inline Neos.Neos/Classes/Controller/Module/Management/WorkspacesController.php]: What if the key zero does not exist?

**Review by markusguenther — APPROVED** [review]
It works fine, and we have just nitpicks in the code.

File: Neos.Neos/Classes/Controller/Module/Management/WorkspacesController.php
Comment (markusguenther) [inline Neos.Neos/Classes/Controller/Module/Management/WorkspacesController.php]: I know it is just a container for the next iteration, but the name doc is bit short... maybe something like `documentsPerDimension`

**Comment by markusguenther** [comment]
**Before:**
Screenshot 2023-03-29 at 14 08 56

**After:**
Screenshot 2023-03-29 at 14 08 27

**Comment by markusguenther** [comment]
Would be nice if someone can have a second look @kdambekalns @mhsdesign @kitsunet 🙏

**Comment by markusguenther** [comment]
No one else available for a review @kitsunet @kdambekalns or @Sebobo 
It would be nice to have a second review, thanks in advance. :)

File: Neos.Neos/Classes/Controller/Module/Management/WorkspacesController.php
Comment (kitsunet) [inline Neos.Neos/Classes/Controller/Module/Management/WorkspacesController.php]: that seems dangerous? I would probably use the dimensionhash?

**Comment by kitsunet** [comment]
I have actually done something like this before in a customer project, I have to compare and remember the edge cases I encountered, so I need a moment, I am pretty sure there were some problems with this (eg. if you create a document in a dimension that also has another dimension as fallback, and then underneath two documents for both dimensions, you shouldn't publish the fallbck sub document as it would miss a parent in that case, stuff like that. this case becomes even worse for content and fallbacks).

File: Neos.Neos/Classes/Controller/Module/Management/WorkspacesController.php
Comment (Jonathan2022Bausch) [reply Neos.Neos/Classes/Controller/Module/Management/WorkspacesController.php]: dimensionhash is not available in postgres DBs yet afaik

Comment (kdambekalns) [reply Neos.Neos/Classes/Controller/Module/Management/WorkspacesController.php]: Must be, if not, that would be a grave bug that should have been noticed.

Comment (Jonathan2022Bausch) [reply Neos.Neos/Classes/Controller/Module/Management/WorkspacesController.php]: I forked neos-development-collection and wrote a migration for it now: https://github.com/neos/neos-development-collection/pull/4429

**Comment by markusguenther** [comment]
Certainly, @kitsunet ! It's great to hear that you have experience with a similar scenario in a customer project. Your insights into potential issues are valuable. To proceed, could you please provide more details about the specific problems you encountered during that project? Understanding the edge cases and challenges you faced will help us address those issues effectively and come up with solutions to ensure a smooth implementation. Please don't hesitate to share any specific scenarios, complications, or concerns you remember encountering. This will enable us to discuss and brainstorm potential solutions to tackle those issues and ensure a successful outcome for your current project.

File: Neos.Neos/Classes/Controller/Module/Management/WorkspacesController.php
Comment (markusguenther) [reply Neos.Neos/Classes/Controller/Module/Management/WorkspacesController.php]: @Jonathan2022Bausch Which versions are you using? Tried it and I have the hash also with Postgres
Screenshot 2023-08-17 at 14 22 39

**Comment by markusguenther** [comment]
Rebased changes so that the PR is mergeable until we find a solution for the discussion :)

File: Neos.Neos/Classes/Controller/Module/Management/WorkspacesController.php
Comment (markusguenther) [reply Neos.Neos/Classes/Controller/Module/Management/WorkspacesController.php]: Ok, understood now the case with the PR https://github.com/neos/neos-development-collection/pull/4429

**Comment by mhsdesign** [comment]
How do we want to proceed here? This will definitely be not easy to upmerge so we need to reimplement this for neos 9 as well.

**Comment by markusguenther** [comment]
In the Neos 9.0 weekly we talked today about the PR and @kitsunet mentioned that he is fine with merging it.

**Review by kitsunet — APPROVED** [review]

**Pull request merged**
